src/Mir/Language.hs

```diff
@@ -91,10 +95,36 @@ runTests (cruxOpts, mirOpts) = do
     let ?assertFalseOnError = True
     let ?printCrucible      = printCrucible mirOpts
 
+    let (filename, nameFilter) = case cargoTestFile mirOpts of
+            -- This case is terrible a hack.  The goal is to mimic the behavior
+            -- of the test binaries produced by `cargo test`, which take a test
+            -- filter string as their main argument, not a filename.  Since we
+            -- can't customize the way Crux parses its non-option arguments, we
+            -- just let it parse its `inputFiles` argument as normal, but we
+            -- treat the value as a test filter instead of a filename.  The
+            -- actual input filename is taken from the `--cargo-test-file`
+            -- option.
+            --
+            -- TODO: Write a proper "cargo-test-like" frontend that does its
+            -- own argument parsing, so we can get rid of this hack and also
+            -- have better `cargo test` compatibility in general.
```

Comment:
Yeah, this seems like it would be very worth doing!

Reply:
Opened #17 for the frontend cleanup